src/java.base/share/classes/sun/security/ssl/NamedGroup.java

```diff
@@ -266,9 +310,19 @@ enum NamedGroup {
         // Check the specific algorithm parameters.
         if (mediator) {
             try {
-                algParams =
-                    AlgorithmParameters.getInstance(namedGroupSpec.algorithm);
-                algParams.init(keAlgParamSpec);
+                // Skip AlgorithmParameters for KEMs (not supported)
+                if (namedGroupSpec == NamedGroupSpec.NAMED_GROUP_KEM) {
```

Comment:
Can you also put in a comment here as to why you're doing these `getInstances`.  

Also, it looks like you're checking for the existence of the `KeyFactory`, but the exception?logger only talks about `AlgorithmParameters`

Reply:
Comment added.